### PR TITLE
Added ROS.18 Validation and conformance suite.

### PR DESCRIPTION
**Pull request by allenbross-wf**
Reason for change
>Rule 18: DPLTurnoverRevenue cannot be a negative value.

Description of change

Added ROS.18
Added conformance suite.

Steps to Test
CI

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/plugin/validate/ROS/rules/__init__.py [hunk 1]
```diff
@@ -17,16 +18,41 @@ def errorOnRequiredFact(
...
+        modelXbrl: ModelXbrl,
+        conceptLn: str,
+        code: str,
+        message: str,
+) -> Iterable[Validation]:
+    """
+    Yields an error if a fact with the given localName is tagged with a negative value.
+    :return: Yields validation errors.
+    """
+    errorModelFacts: list[ModelFact | None] = []
```
Comment (derekgengenbacher-wf) [inline arelle/plugin/validate/ROS/rules/__init__.py:47]: I dont believe that you need a None here for typing. The list will either be a list of of model facts or an empty list of model facts.

File: arelle/plugin/validate/ROS/rules/__init__.py [hunk 2]
```diff
@@ -17,16 +18,41 @@ def errorOnRequiredFact(
...
 ) -> Iterable[Validation]:
     """
-    Yields an error if a fact with the given QName is not tagged with a non-nil value.
+    Yields an error if a fact with the given localName is not tagged with a non-nil value.
     :return: Yields validation errors.
     """
     facts: set[ModelFact] = modelXbrl.factsByLocalName.get(conceptLn, set())
     for fact in facts:
         if not fact.isNil:
             return
```
Comment (derekgengenbacher-wf) [inline arelle/plugin/validate/ROS/rules/__init__.py:27]: If a list of facts gets run through this we do not indicate which facts are actually in error. It seems like this should be appending fats to a list and then returning that list as the model objects of the error?

Comment (aaroncameron-wk) [reply arelle/plugin/validate/ROS/rules/__init__.py:27]: I think the function name may be misleading. The intent is to error on a _missing_ required fact. There are no facts to include in the error, just the concept for which a fact is missing. Suggest `errorOnMissingRequiredFact`?

File: arelle/plugin/validate/ROS/rules/__init__.py [hunk 3]
```diff
@@ -17,16 +18,41 @@ def errorOnRequiredFact(
...
+        conceptLn=conceptLn,
         codes=code,
         msg=message,
-        conceptLn=conceptLn,
+
 
     )
+
+
+def errorOnPositiveOnlyFact(
```
Comment (aaroncameron-wk) [inline arelle/plugin/validate/ROS/rules/__init__.py:37]: I think the name here is the opposite of the implementation. It errors on a negative fact. Suggest `errorOnNegativeFact`.

File: arelle/plugin/validate/ROS/rules/__init__.py (hunk 2 above)
Comment (allenbross-wf) [reply arelle/plugin/validate/ROS/rules/__init__.py:27]: I don't think that's necessarily wrong to have a fact with a nil-value , the rule seems to indicate that one fact needs to be non-nil.  Not all of them.

**Review by derekgengenbacher-wf — APPROVED** [review]

**Review by aaroncameron-wk — APPROVED** [review]

**Pull request merged**